### PR TITLE
add a wrapper method which waits until the data is valid

### PR DESCRIPTION
**Pull request by 0xCCD**
this makes the mentioned workaround (adding sleep) unnecessary and solves all problems described in the issues #3 #4 #5 #6

### CODE REVIEW

**Comment by byWulf** [comment]
Just got a chance to test this out. Works for me! :) 

Would like the following changes to the code:
* Exit the while-loop when the timeout is reached
* stop the timeout when `resolve` is called (otherwise `reject` will also get called everytime in your code)
* code formatting (4 spaces as indent)
* perhaps use async/await instead of the "old" promise
Like so (untested):
```
    this.wait_until_configuration_is_finished = async (args, timeout=3000) => {
        let run = true;
        const timeout = setTimeout( () => {
            run = false;
        }, timeout);

        while(run) {
            const reply = await this.spi_transfer_array(args);
            if(reply[5] === this.SENSOR_STATE.VALID_DATA){
                clearTimeout(timeout);
                return reply;
            }
        }

        throw new Error('timeout reached');
    }
```

Could you then also revert the change in my commit [eecfaa](https://github.com/byWulf/brickpi3-nodejs/commit/eecfaac7a4c714af8cb4e8227937bf12eff45ba5), where I added the sleep to bypass the bug?

After that I would love to merge your PR. :)

Thanks!

**Comment by 0xCCD** [comment]
Hi, I've finished your change requests except the **perhaps use async/await instead of the "old" promise** part, since you have massively used Promise in this file and did not used any await or async there. I have also extended the timeout handling a bit and added the ability to overwrite the default of 3000ms for some special cases. While testing, the timeout handling worked now as expected.

File: index.js [hunk 1]
```diff
@@ -585,6 +585,31 @@ function BrickPi3(address = 1) {
...
+                const msg = `timeout reached: sensor configuration not successfully within ${configurationTimeout}ms`;
+                run = false;
+                throw new Error(msg);
+            }, configurationTimeout);
+
+            while(run) {
+                const reply = await this.spi_transfer_array(args);
+                if(reply[5] === this.SENSOR_STATE.VALID_DATA){
+                    clearTimeout(timeout);
+                    return resolve(this.spi_transfer_array(args));
```
Comment (byWulf) [inline index.js]: You can resolve directly with `reply` without the need of calling `spi_transfer_array` again.

**Comment by byWulf** [comment]
Fair point with the async/await! Is okay for me :)

**Comment by byWulf** [comment]
Thanks for your contribution! :) I release 0.9.8 with your fix and closed the corresponding issues.

**Pull request merged**